### PR TITLE
[#233] Bump Haskell dependencies and remove Lorentz version

### PR DESCRIPTION
**Pull request by pasqu4le**
## Description

Bumps Haskell dependencies to a newer versions.

Moreover it removes the outdated Lorentz version of the contract entirely.

## Related issue(s)

Part of #233 

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

[//]: # (Add more docs here if you have them in the repository)
- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock

#### Stylistic guide (mandatory)

- [x] My commits comply with [the following policy](https://www.notion.so/serokell/Commit-and-PR-policy-4cf98e1a910a415d86b5f2491d9af1af).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

File: haskell/src/Ligo/Typescript.hs [hunk 1]
```diff
@@ -315,6 +316,9 @@ mkTypesFor typename epType = case U.unwrapT epType of
   U.TBls12381G1 -> [TsType typename (TsAlias TsString)]
   U.TBls12381G2 -> [TsType typename (TsAlias TsString)]
   U.TNever -> [TsType typename (TsAlias TsVoid)]
+  U.TTicket a ->
+    let ticketFields = [("ticketer", toType U.TAddress), ("value", a), ("amount", toType U.TNat)]
+    in [TsType typename (TsInterface $ map (fst . mkTypesForField False) ticketFields)]
```
Comment (sras) [inline haskell/src/Ligo/Typescript.hs:321]: Here you seem to be throwing away the `[TsDecl]` returned by  the `mkTypesForField`. This should be fine for the first and last elements for `ticketFields` as they are primitives. But for the second field, the type is `a`, which could require additional declarations. So if we throw the `[TsDecl]` away, the nested declarations for `a` won't be included. So I think something like the following might be required here (taken from the TPair branch of the same function.

```
        decls = (mkTypesForField False) <$> (indexEmptyFields allFields)
        recordFields = fst <$> decls
        thisType = TsType typename (TsInterface recordFields)
        thisDecls = snd <$> decls
        in thisType : (concat thisDecls)
```

```
        decls = (mkTypesForField False) <$> (indexEmptyFields allFields)
        recordFields = fst <$> decls
        thisType = TsType typename (TsInterface recordFields)
        thisDecls = snd <$> decls
        in thisType : (concat thisDecls)
```

**Review by sras — APPROVED** [review]
LGTM

**Comment by rinn7e** [comment]
I couldn't build this locally, due to this error:
```
[1 of 6] Compiling Ligo.BaseDAO.Types
<command line>: /home/rinne/.stack/snapshots/x86_64-linux-tinfo6/9f1ebce10328f480dad134acff21eed56118d6ab46429e4dbe02498c01f0293a/8.10.4/lib/x86_64-linux-ghc-8.10.4/libHSmorley-metadata-0-Gk5LOjJkrb7FeQfRlzMWR2-ghc8.10.4.so: undefined symbol: lorentzzzm0zi11zi0zmDlcSfoFCsgnH10lVQsgBTs_LorentzzziMacro_unpair2_closure
```
Not sure how to fix this.

**Comment by pasqu4le** [comment]
@rinn7e oddy I had a similar problem, not sure why but it's an issue with stack.

After I cleaned (part of) my `.stack` dir it recompiled and all worked correctly, I'm afraid you'll need to do the same.

**Comment by rinn7e** [comment]
> @rinn7e oddy I had a similar problem, not sure why but it's an issue with stack.
> 
> After I cleaned (part of) my `.stack` dir it recompiled and all worked correctly, I'm afraid you'll need to do the same.

I have to nuke the entire dir, but it is working fine now.

**Review by rinn7e — APPROVED** [review]

**Review by sras — APPROVED** [review]

**Pull request merged**